specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntityQueries.json

```diff
@@ -0,0 +1,651 @@
+        },
+        "inputEntityType": {
+          "$ref": "#/definitions/EntityInnerType",
+          "description": "The type of the query's source entity"
+        },
+        "inputFields": {
+          "description": "List of the fields of the source entity that are required to run the query",
+          "items": {
+            "description": "input field",
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "outputEntityTypes": {
+          "description": "List of the desired output types to be constructed from the result",
+          "items": {
+            "$ref": "#/definitions/EntityInnerType",
+            "description": "output entity type"
+          },
+          "type": "array"
+        },
+        "queryTemplate": {
+          "description": "The template query string to be parsed and formatted",
+          "type": "string"
+        }
+      },
+      "type": "object"
+    },
+    "ActivityEntityQuery": {
+      "description": "Represents Activity entity query.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/EntityQuery"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ActivityEntityQueriesProperties",
+          "description": "Activity entity query properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "Activity"
+    },
+    "ActivityCustomEntityQuery": {
+      "description": "Represents Activity entity query.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CustomEntityQuery"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ActivityEntityQueriesProperties",
+          "description": "Activity entity query properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "Activity"
+    },
+    "ActivityEntityQueriesProperties": {
+      "description": "Describes activity entity query properties",
+      "properties": {
+        "title": {
+          "description": "The entity query title",
+          "type": "string"
+        },
+        "content": {
+          "description": "The entity query content to display in timeline",
+          "type": "string"
+        },
+        "description": {
+          "description": "The entity query description",
+          "type": "string"
+        },
+        "queryDefinitions": {
+          "description": "The Activity query definitions",
+          "properties": {
+            "query": {
+              "description": "The Activity query to run on a given entity",
+              "type": "string"
+            }
+          },
+          "type": "object"
+        },
+        "inputEntityType": {
+          "$ref": "#/definitions/EntityInnerType",
+          "description": "The type of the query's source entity"
+        },
+        "requiredInputFieldsSets": {
```

Comment:
Please add an example for PUT that contains this property.  If this is a doubly nested array it will be much easier to work with as an array of objects that contain an array property.  I.e. `"requiredInputFieldSets": [ { "fields": [] }, { "fields": [] } ]`.  I've seen some json parsers have problems with nested arrays in the past

Reply:
I did, see "examples/entityQueries/CreateEntityQueryActivity.json"
it is a doubly nested array such as:
[
["fiels1"],
["field2","field3"]
]